std/xml.d

```diff
@@ -1059,9 +1059,13 @@ class Tag
                 munch(s,whitespace);
                 reqc(s,'=');
                 munch(s,whitespace);
-                reqc(s,'"');
-                string val = decode(munch(s,"^\""), DecodeMode.LOOSE);
-                reqc(s,'"');
+                string atr_val_put_chars="'\"";   //chars in which tags can be put into
+                char atr_put_into = reqs(s,atr_val_put_chars);
+                char[] r=new char[2];
+                r[0] = '^';
+                r[1] = atr_put_into;
```

Comment:
As far as I can tell, this is a lengthy way of doing essentially `string r = "^" ~ atr_put_into;`. I'd say either keep it simple and use that one, or use `char[2] r = ['^', atr_put_into];` to avoid the allocation. Also, how about a more expressive name for `r`? `notQuote` maybe?